mlflow/metrics/genai/genai_metric.py

```diff
@@ -244,6 +245,13 @@ def score_model_on_one_payload(
                 raw_result = model_utils.score_model_on_payload(eval_model, payload)
                 return _extract_score_and_justification(raw_result)
             except Exception as e:
+                open_ai_error_header = "Error response from OpenAI as follows:\n "
```

Comment:
I am curious if we should remove this try...catch instead so we can throw all the errors. Example gateway key/route is not set correctly. OpenAi key is incorrect.
Actually, we shouldn't remove the try...catch since we have None, None returned at the end for cases where we couldn't parse the response correctly. 

I think here, from open_ai and gateway we can throw a particular error_type from MlflowException that we can check here?

@harupy Any thoughts or suggestions here?

Reply:
> we can throw a particular error_type from MlflowException that we can check here?

+1